### DISCUSSION TITLE
Pub Meeting @ The Alex (3h)

### DISCUSSION THREAD

**Discussion by richardbirkin** [SuperHyperInstantFutureTime/TrackShift#55] [Meetings]
3h @richardbirkin @g105b 

We sat next to the pub rabbit and enjoyed some Pentrich ales. 

We talked about badges and mission patches and possible naming conventions for significant releases/company milestones. We don't want to be metaphorical because it becomes meaningless (what is Apollo 9 when talking about Trackshift?). 

Also - we like the style of pump clips at pubs. We like meeting in the pub so this feels very us.

Also - we know a brewery owner so could get Super Hyper Instant Future Time or Trackshift beer to celebrate a significant achievement. Speaking of achievements...

## Milestone 3 Completion

We have a working deployment pipeline. We watched the tests run in github, then went to our dev URL and saw Trackshift on a server on the real internet! 

Greg added `gt build` to the deployment process, but he hates YAML and the tests failed due to something to do with SASS. It took an hour to fix, but he did it. 

It was worth brining the laptops to see the deployment working, but we'll try not to do that again as it didn't feel very productive. The upside is that the deployment process is now futureproof. 

Greg then did some quick troubleshooting on my laptop to get Trackshfit working locally for me. With this complete we closed Milestone 3.  

## Milestone 4

We talked about how the Scenarios for artists that made up Milestones 1 to 3 would work for Labels. 

I'd never heard about 'duck code' before. If it looks like a duck, quacks like a duck and swims like a duck, it's a duck. 

This is our approach to identifying what type of Distribution Statement is being uploaded. 

![IMG_6192](https://user-images.githubusercontent.com/115098450/206436090-f5950287-e164-4ad4-963f-f862ddae6d36.jpg)

This is my diagram to show Greg the relationship between Artists, Labels and Vendors. He was confused. It is confusing. 

Hopefully I can explain it bit by bit in Scenarios when I start writing up Issues for Milestone 4.

## Tangents

We went on a few tangents about keeping our code Open Source, why PHP is awesome (it will just do stuff), and security issues encountered by Greg when working for clients in the medical sector. 

On the latter point, this is why our Board of Advisors is going to be important - to challenge us on things that we might not have thought of or experienced before.

**Comment by g105b** [comment]
Some follow up notes.

Milestone 3 was rewarding to see go live. For the record, the dev server can be reached at https://dev.trackshift.app, defaulting to the master branch, but any branch that's live can be reached with branch-name.dev.trackshift.app.

Something that hasn't been done yet is to share the same data between all dev branches - it might be best that way, but I have a feeling that we'll want to see our existing data from one branch within another... maybe not, I don't know yet. It's something we can discuss next time.

Another thing that isn't done yet is deployment of releases, because this will use Digital Ocean's API ([doctl](https://docs.digitalocean.com/reference/doctl/)) to spin up a fresh server for each release - there's a few moving parts there (like persisting data between releases, handling users that are in the app mid-release, etc.) and [I need to start jotting some notes down here](https://github.com/SuperHyperInstantFutureTime/Trackshift/issues/56).

The mac setup guide was flawed - I wrote the guide in the notes app on my mac, and didn't realise it changed `"` characters to `‟`, which broke when Biff pasted into his terminal. Works a treat now though.

YAML ... YUCK! But I did read that YAML is a superset of JSON, so apparently it's possible to translate YAML to JSON, and it should work with "real" syntax. I might give it a go next time I'm wrestling the YAML config. For the record, a whole hour was wasted trying to configure YAML. For a laugh, https://noyaml.com/

The beer and the company was top notch as usual. Now I'm going to stare at some label files with a cup of tea, to see if I can get anything to sink in while my brain's still got 5% charge tonight.

**Reply by richardbirkin** [reply]
Beer + Company being top notch >>> seconded. Worth noting the beers were made by Pentrich and I think we sampled them all. At least all the pales on offer of which there were three. The favourite was 'Pot Luck Happy'

**Reply by g105b** [reply]
Oh yeah, if we're fully open source we need to be open about our sauce!

**Comment by richardbirkin** [comment]
I just remembered we also talked about Winamp being amazing...and how we want to do a side project that makes a really visual media library out of a folder of mp3s where you can order things like you would your CD/record collection.
